lib/src/metta/runner/builtin_mods/fileio.rs

```diff
@@ -0,0 +1,156 @@
+use std::cell::RefCell;
+use std::fmt::{Display, Formatter};
+use crate::{sym, Atom, ExecError, Grounded};
+use crate::metta::{ARROW_SYMBOL, UNIT_ATOM};
+use crate::metta::runner::stdlib::unit_result;
+use crate::metta::runner::str::{Str, ATOM_TYPE_STRING};
+use std::fs;
+use std::fs::OpenOptions;
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::rc::Rc;
+use crate::space::grounding::GroundingSpace;
+use crate::metta::text::SExprParser;
+use crate::metta::runner::{ModuleLoader, RunContext, DynSpace, Metta, MettaMod};
+use crate::atom::gnd::*;
+
+pub static FILEIO_METTA: &'static str = include_str!("fileio.metta");
+pub const ATOM_TYPE_FILE_HANDLE: Atom = sym!("FileHandle");
+
+#[derive(Clone, Debug)]
+pub struct FileHandle(Rc<RefCell<fs::File>>);
+
+impl FileHandle {
+    fn open(path: &str, options: &str) -> Result<Self, ExecError> {
+
+        let mut opened_file = OpenOptions::new();
+        let opened_file= opened_file.read(options.contains("r")).write(options.contains("w")).create(options.contains("c")).append(options.contains("a")).truncate(!options.contains("a"));
```

Comment:
It is more readable in this way:
```suggestion
        let opened_file = opened_file.read(options.contains("r"))
            .write(options.contains("w"))
            .create(options.contains("c"))
            .append(options.contains("a"))
            .truncate(!options.contains("a"));
```